src/utils/async.ts

```diff
@@ -16,6 +16,24 @@ export const mapAsyncIterator = <T, V>(
     )
 });
 
+/**
+ * Async maps over an AsyncIterator
+ */
+export const asyncMapAsyncIterator = <T, V>(
```

Comment:
I argue we don't need it, we can just add an `await` to `mapAsyncIterator`

Reply:
Tried it, but it was not resolving nested promises.